spec/integration/util/windows/user_spec.rb

```diff
@@ -155,6 +156,20 @@ def expect_logon_failure_error(&block)
       end
     end
 
+    describe 'logon_user_by_logon_type' do
+      let(:password){'test'}
+      let(:fLOGON32_LOGON_INTERACTIVE){2}
+      let(:fLOGON32_LOGON_NETWORK){3}
+      let(:fLOGON32_PROVIDER_DEFAULT){0}
+      let(:token){'test_token'}
+      it 'should raise error given that logon returns false' do
+        allow(Puppet::Util::Windows::User.logon_user_by_logon_type).to receive(:LogonUserW).and_return(false)
```

Comment:
You will need to use `recieve().with()` to define a set of parameters to `:LogonUserW` to make the CI errors go away.

Reply:
That's not quite the problem here. The CI failure is because `allow(Puppet::Util::Windows::User.logon_user_by_logon_type)` is trying to call `Puppet::Util::Windows::User.logon_user_by_logon_type` without any arguments. This should be `allow(Puppet::Util::Windows::User).to receive(:LogonUserW).and_return(false)` since you can't really easily say "expect method X to be called, but _only_ from within method Y".